File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
         
        HashMap<Integer, Integer> map = new HashMap<>();
        PriorityQueue<Integer> minHeap = new PriorityQueue<>((a,b) -> map.get(a) - map.get(b));
        for(int num : nums){
            map.put(num, map.getOrDefault(num, 0) + 1);   
        }

        for(Integer key : map.keySet()){
            minHeap.add(key);
            if(minHeap.size() > k){
                minHeap.poll();
            }
        }

        int[] result = new int[k];
        for (int i = k - 1; i >= 0; i--) {
            result[i] = minHeap.poll(); 
        }
        return result;


        
        
    }
}

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-0.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:

        prefix = 0
        hmap = {0:1}
        result = 0
        for num in nums:
            prefix+=num
            if prefix - k in hmap:
                result += hmap[prefix - k]
            hmap[prefix] = hmap.get(prefix, 0) + 1
        
        return result

        